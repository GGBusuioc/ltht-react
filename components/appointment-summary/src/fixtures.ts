import { Encounter, EncounterStatusCode, PartialDateTimeKindCode, Metadata } from '@ltht-react/types'

const mockMetadata: Metadata = {
  dataSources: [
    {
      display: 'Mock',
    },
  ],
}

const AppointmentOne: Encounter = {
  id: '346ca927-67e0-e931-a2c7-115087226fg2|appointment',
  metadata: mockMetadata,
  type: [
    {
      text: 'See GP regarding sore throat',
      coding: [{ display: 'GP', code: '64706' }],
    },
  ],
  period: {
    start: {
      value: '2020-04-01T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Date,
    },
  },
  serviceProvider: {
    display: 'Beeston Village Surgery',
    typeName: 'GP',
  },
  status: EncounterStatusCode.Planned,
}

const AppointmentTwo: Encounter = {
  id: '941ca927-34e0-e911-a2c7-005087226fe4|appointment',
  metadata: mockMetadata,
  type: [
    {
      text: 'Chest x-ray for recent car accident',
      coding: [{ display: 'Hospital', code: '26706' }],
    },
  ],
  period: {
    start: {
      value: '2019-12-12T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Date,
    },
  },
  serviceProvider: {
    display: 'Chapel Allerton Hospital',
    typeName: 'Hospital',
  },
  status: EncounterStatusCode.Finished,
}

export { AppointmentOne, AppointmentTwo }
