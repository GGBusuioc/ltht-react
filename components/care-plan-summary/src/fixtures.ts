import { CarePlan, CarePlanStatusCode, PartialDateTimeKindCode, CarePlanIntentCode, Metadata } from '@ltht-react/types'

const mockMetadata: Metadata = {
  dataSources: [
    {
      display: 'Mock',
    },
  ],
}

const CarePlanOne: CarePlan = {
  id: '580ca927-34e0-e911-a2c7-005056926fe4|careplan',
  metadata: mockMetadata,
  author: { display: 'Joe Bloggs', type: 'Test' },
  intent: CarePlanIntentCode.Plan,
  period: {
    start: {
      value: '2013-01-01T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Date,
    },
    end: {
      value: '2014-02-02T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Date,
    },
  },
  status: CarePlanStatusCode.Active,
  title: 'Test Care Plan Summary One',
}

const CarePlanTwo: CarePlan = {
  id: '690ca927-68e0-c511-b2c7-898056926ze4|careplan',
  metadata: mockMetadata,
  author: { display: 'John Mclane', type: 'Test' },
  intent: CarePlanIntentCode.Proposal,
  period: {
    start: {
      value: '2014-02-02T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Date,
    },
    end: {
      value: '2015-03-04T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Date,
    },
  },
  status: CarePlanStatusCode.Completed,
  title: 'Test Care Plan Summary Two',
}

export { CarePlanOne, CarePlanTwo }
