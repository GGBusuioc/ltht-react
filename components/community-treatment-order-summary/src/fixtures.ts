import {
  LypftCommunityTreatmentOrder,
  CarePlanStatusCode,
  PartialDateTimeKindCode,
  CarePlanIntentCode,
  Metadata,
} from '@ltht-react/types'

const mockMetadata: Metadata = {
  dataSources: [
    {
      display: 'Mock',
    },
  ],
}

const CommunityTreatmentOrderOne: LypftCommunityTreatmentOrder = {
  id: '580ca927-34e0-e911-a2c7-005056926fe4|communityTreatmentOrder',
  metadata: mockMetadata,
  consentToTreat: 'Consent Not Given',
  legalStatus: {
    coding: [{ display: 'Approved', code: '3309007' }],
    text: 'Approved',
  },
  period: {
    start: {
      value: '2013-01-01T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Date,
    },
    end: {
      value: '2014-02-02T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Date,
    },
  },
  restrictions: 'No restrictions',
}

const CommunityTreatmentOrderTwo: LypftCommunityTreatmentOrder = {
  id: '690ca927-68e0-c511-b2c7-898056926ze4|communityTreatmentOrder',
  metadata: mockMetadata,
  consentToTreat: 'Consent Given',
  legalStatus: {
    coding: [{ display: 'Pending', code: '779007' }],
    text: 'Pending',
  },
  period: {
    start: {
      value: '2020-01-02T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Date,
    },
  },
  restrictions: 'No Restrictions',
}

export { CommunityTreatmentOrderOne, CommunityTreatmentOrderTwo }
