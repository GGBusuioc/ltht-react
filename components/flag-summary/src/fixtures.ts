import { Flag, FlagStatusCode, PartialDateTimeKindCode, Metadata } from '@ltht-react/types'

const mockMetadata: Metadata = {
  dataSources: [
    {
      display: 'Mock',
    },
  ],
}

const FlagOne: Flag = {
  id: '580ca927-34e0-e911-a2c7-005056926fe4|personalertandhazard',
  metadata: mockMetadata,
  author: {
    display: 'John Smith',
    reference:
      'https://phoenixqa.careworks.ie/Services/api/fhir/v3/practitioner/9c1ad2c0-ec34-e911-a2c5-005056926fe4|systemuser',
    type: 'Practitioner',
    identifier: undefined,
  },
  code: {
    coding: [{ display: 'Lives in Known High Risk Area', code: '109006' }],
  },
  period: {
    start: {
      value: '2013-01-01T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Year,
    },
  },
  status: FlagStatusCode.Inactive,
}

const FlagTwo: Flag = {
  id: '580ca927-34e0-e911-a2c7-005056926fe4|personalertandhazard',
  metadata: mockMetadata,
  author: {
    display: 'John Smith',
    reference:
      'https://phoenixqa.careworks.ie/Services/api/fhir/v3/practitioner/9c1ad2c0-ec34-e911-a2c5-005056926fe4|systemuser',
    type: 'Practitioner',
    identifier: undefined,
  },
  code: {
    coding: [{ display: 'High Risk of Violence', code: '56706' }],
  },
  category: {
    coding: [{ display: 'Violence', code: '66706' }],
  },
  period: {
    start: {
      value: '2013-01-01T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Date,
    },
    end: {
      value: '2014-02-02T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Date,
    },
  },

  status: FlagStatusCode.Inactive,
}

export { FlagOne, FlagTwo }
