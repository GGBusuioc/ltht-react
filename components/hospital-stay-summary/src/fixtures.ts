import { HospitalStay, PartialDateTimeKindCode, Metadata } from '@ltht-react/types'

const mockMetadata: Metadata = {
  dataSources: [
    {
      display: 'Mock',
    },
  ],
}

const HospitalStayOne: HospitalStay = {
  id: '580ca927-34e0-e911-a2c7-005056926fe4|hospitalstay',
  metadata: mockMetadata,
  nhsNumber: '123456789',
  serviceProvider: { display: 'Chapel Allerton Hospital', type: 'Hospital' },
  period: {
    start: {
      value: '2013-01-01T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Date,
    },
    end: {
      value: '2014-02-02T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Date,
    },
  },
}

const HospitalStayTwo: HospitalStay = {
  id: '690ca927-68e0-c511-b2c7-898056926ze4|hospitalstay',
  metadata: mockMetadata,
  nhsNumber: '987654321',
  serviceProvider: { display: 'Beaumont Hospital', type: 'Hospital' },
  period: {
    start: {
      value: '2020-01-01T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Date,
    },
  },
}

export { HospitalStayOne, HospitalStayTwo }
