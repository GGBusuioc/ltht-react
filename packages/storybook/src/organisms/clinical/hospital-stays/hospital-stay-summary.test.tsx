import React from 'react'
import { mount } from 'enzyme'
import ReactDOM from 'react-dom'

import HospitalStaySummary from '@ltht-react/hospital-stay-summary'
import { HospitalStayOne, HospitalStayTwo } from './hospital-stay.fixtures'

describe('HospitalStaySummary', () => {
  const hospitalStays = [HospitalStayOne, HospitalStayTwo]
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<HospitalStaySummary hospitalStays={hospitalStays} />, div)
  })
  it('matches snapshot', () => {
    expect(mount(<HospitalStaySummary hospitalStays={hospitalStays} />)).toMatchSnapshot('wrapper mount')
  })
})
