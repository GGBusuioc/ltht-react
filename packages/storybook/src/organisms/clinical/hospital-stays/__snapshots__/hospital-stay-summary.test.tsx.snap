// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HospitalStaySummary matches snapshot: wrapper mount 1`] = `
.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.emotion-2 > div {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.emotion-0 {
  color: #231f20;
  text-align: left;
}

.emotion-1 {
  color: -v-a-l-u-e:#768692;
  -d-a-r-k-e-r25: #59656e;
  -l-i-g-h-t-e-r25: #98a4ad;
  text-align: right;
}

<HospitalStaySummary
  hospitalStays={
    Array [
      Object {
        "id": "580ca927-34e0-e911-a2c7-005056926fe4|hospitalstay",
        "metadata": Object {
          "dataSources": Array [
            Object {
              "display": "Mock",
            },
          ],
          "isRedacted": true,
          "requestedWhen": "",
        },
        "period": Object {
          "end": Object {
            "kind": "DATE",
            "value": "2014-02-02T00:00:00+00:00",
          },
          "start": Object {
            "kind": "DATE",
            "value": "2013-01-01T00:00:00+00:00",
          },
        },
        "serviceProvider": Object {
          "display": "Chapel Allerton Hospital",
          "typeName": "Hospital",
        },
      },
      Object {
        "id": "690ca927-68e0-c511-b2c7-898056926ze4|hospitalstay",
        "metadata": Object {
          "dataSources": Array [
            Object {
              "display": "Mock",
            },
          ],
          "isRedacted": true,
          "requestedWhen": "",
        },
        "period": Object {
          "start": Object {
            "kind": "DATE",
            "value": "2020-01-01T00:00:00+00:00",
          },
        },
        "serviceProvider": Object {
          "display": "Beaumont Hospital",
          "typeName": "Hospital",
        },
      },
    ]
  }
>
  <ListItem
    key="580ca927-34e0-e911-a2c7-005056926fe4|hospitalstay"
  >
    <li>
      <HospitalStaySummaryItem
        hospitalStay={
          Object {
            "id": "580ca927-34e0-e911-a2c7-005056926fe4|hospitalstay",
            "metadata": Object {
              "dataSources": Array [
                Object {
                  "display": "Mock",
                },
              ],
              "isRedacted": true,
              "requestedWhen": "",
            },
            "period": Object {
              "end": Object {
                "kind": "DATE",
                "value": "2014-02-02T00:00:00+00:00",
              },
              "start": Object {
                "kind": "DATE",
                "value": "2013-01-01T00:00:00+00:00",
              },
            },
            "serviceProvider": Object {
              "display": "Chapel Allerton Hospital",
              "typeName": "Hospital",
            },
          }
        }
      >
        <div
          className="emotion-2"
          role="link"
        >
          <div>
            <HospitalStayPeriod
              encounter={
                Object {
                  "id": "580ca927-34e0-e911-a2c7-005056926fe4|hospitalstay",
                  "metadata": Object {
                    "dataSources": Array [
                      Object {
                        "display": "Mock",
                      },
                    ],
                    "isRedacted": true,
                    "requestedWhen": "",
                  },
                  "period": Object {
                    "end": Object {
                      "kind": "DATE",
                      "value": "2014-02-02T00:00:00+00:00",
                    },
                    "start": Object {
                      "kind": "DATE",
                      "value": "2013-01-01T00:00:00+00:00",
                    },
                  },
                  "serviceProvider": Object {
                    "display": "Chapel Allerton Hospital",
                    "typeName": "Hospital",
                  },
                }
              }
            >
              <div
                className="emotion-0"
              >
                01-Jan-2013 to 02-Feb-2014
              </div>
            </HospitalStayPeriod>
          </div>
          <div>
            <HospitalStayServiceProvider
              encounter={
                Object {
                  "id": "580ca927-34e0-e911-a2c7-005056926fe4|hospitalstay",
                  "metadata": Object {
                    "dataSources": Array [
                      Object {
                        "display": "Mock",
                      },
                    ],
                    "isRedacted": true,
                    "requestedWhen": "",
                  },
                  "period": Object {
                    "end": Object {
                      "kind": "DATE",
                      "value": "2014-02-02T00:00:00+00:00",
                    },
                    "start": Object {
                      "kind": "DATE",
                      "value": "2013-01-01T00:00:00+00:00",
                    },
                  },
                  "serviceProvider": Object {
                    "display": "Chapel Allerton Hospital",
                    "typeName": "Hospital",
                  },
                }
              }
            >
              <div
                className="emotion-1"
              >
                Chapel Allerton Hospital
              </div>
            </HospitalStayServiceProvider>
          </div>
        </div>
      </HospitalStaySummaryItem>
    </li>
  </ListItem>
  <ListItem
    key="690ca927-68e0-c511-b2c7-898056926ze4|hospitalstay"
  >
    <li>
      <HospitalStaySummaryItem
        hospitalStay={
          Object {
            "id": "690ca927-68e0-c511-b2c7-898056926ze4|hospitalstay",
            "metadata": Object {
              "dataSources": Array [
                Object {
                  "display": "Mock",
                },
              ],
              "isRedacted": true,
              "requestedWhen": "",
            },
            "period": Object {
              "start": Object {
                "kind": "DATE",
                "value": "2020-01-01T00:00:00+00:00",
              },
            },
            "serviceProvider": Object {
              "display": "Beaumont Hospital",
              "typeName": "Hospital",
            },
          }
        }
      >
        <div
          className="emotion-2"
          role="link"
        >
          <div>
            <HospitalStayPeriod
              encounter={
                Object {
                  "id": "690ca927-68e0-c511-b2c7-898056926ze4|hospitalstay",
                  "metadata": Object {
                    "dataSources": Array [
                      Object {
                        "display": "Mock",
                      },
                    ],
                    "isRedacted": true,
                    "requestedWhen": "",
                  },
                  "period": Object {
                    "start": Object {
                      "kind": "DATE",
                      "value": "2020-01-01T00:00:00+00:00",
                    },
                  },
                  "serviceProvider": Object {
                    "display": "Beaumont Hospital",
                    "typeName": "Hospital",
                  },
                }
              }
            >
              <div
                className="emotion-0"
              >
                01-Jan-2020 to Present
              </div>
            </HospitalStayPeriod>
          </div>
          <div>
            <HospitalStayServiceProvider
              encounter={
                Object {
                  "id": "690ca927-68e0-c511-b2c7-898056926ze4|hospitalstay",
                  "metadata": Object {
                    "dataSources": Array [
                      Object {
                        "display": "Mock",
                      },
                    ],
                    "isRedacted": true,
                    "requestedWhen": "",
                  },
                  "period": Object {
                    "start": Object {
                      "kind": "DATE",
                      "value": "2020-01-01T00:00:00+00:00",
                    },
                  },
                  "serviceProvider": Object {
                    "display": "Beaumont Hospital",
                    "typeName": "Hospital",
                  },
                }
              }
            >
              <div
                className="emotion-1"
              >
                Beaumont Hospital
              </div>
            </HospitalStayServiceProvider>
          </div>
        </div>
      </HospitalStaySummaryItem>
    </li>
  </ListItem>
</HospitalStaySummary>
`;
